src/providers/FusionAuthProvider.tsx

```diff
@@ -62,25 +82,14 @@ enum FunctionType {
     register = 'register',
 }
 
-async function generateUrl(
+function generateUrl(
     functionType: FunctionType,
     baseUrl: string,
-    clientID: string,
-    scope: string,
-    redirectUri: string,
-    state = '',
+    queryString: Record<string, string>,
 ) {
-    const queryString = new URLSearchParams({
-        client_id: clientID,
-        scope: scope,
-        response_type: 'code',
-        redirect_uri: redirectUri,
-        code_challenge: await generatePKCE(),
-        code_challenge_method: 'S256',
-        state: `${generateRandomString()}:${state}`,
-    });
-
-    return `${baseUrl}/${functionType}?${queryString}`;
+    const query = new URLSearchParams(queryString);
+
+    return `${baseUrl}/${functionType}?${query}`;
```

Comment:
Can we move this function inside the provider? Since we are passing in function type, it is already coupled to the provider, and so if we move it we won't need to pass the baseUrl into the function call as it will know through state.

This will mean it has to be stabilized with `useCallback`